docs/advanced/pycpp/numpy.rst

```diff
@@ -37,7 +37,7 @@ completely avoid copy operations with Python expressions like
        .def_buffer([](Matrix &m) -> py::buffer_info {
             return py::buffer_info(
                 m.data(),                               /* Pointer to buffer */
-                sizeof(float),                          /* Size of one scalar */
+                (py::ssize_t)sizeof(float),             /* Size of one scalar */
```

Comment:
Are you sure an explicit cast is necessary here? As far as I know, compilers don't issue warnings for compile-time constants which are guaranteed to fit into the destination type:

```c++
#include <cstddef>

void foo(int) {}

int main() {
    foo(sizeof(float)); // clean (no warnings)
    size_t s = 4;
    foo(s);             // <-- conversion warning here
}
```

GCC, clang and MSVC for example: https://godbolt.org/g/zgibas

This might help avoid some of the casts on the user side.